### PR TITLE
Add attribution metadata to asset catalogue

### LINKED ISSUE #1620: Add attribution metadata to asset catalogue
1) Add attribution metadata to asset catalogue consisting of 
- Asset attribution (text)
- Asset source (url)
- Asset licence
None are mandatory

2) Add plugin to authoring tool to auto fill the attribution field of components that have it when selecting an asset.

### PR DESCRIPTION
**Pull request by davetaz**
closes #1620 

## Proposed changes

This set of changes adds three fields to the asset catalogue so that users can properly record the source, attribution and licence for using assets. 

These have been added to the data model, interfaces in the authoring tool as well as the import/export plugins. 

## Checklist

Below is a checklist of the key things that the core team will be looking for when reviewing your PR. Please check off as many of these as you can prior to submitting the PR, but don't worry if you can't do all of them, the core team are here to help!

- [Y] I have read the [CONTRIBUTING](https://github.com/adaptlearning/adapt_authoring/blob/master/.github/CONTRIBUTING.md) doc
- [Y] I have checked that the code compiles correctly
- [Y] I have checked that unit test suite passes locally with my changes
- [n/a] I have added new tests that cover my changes (where appropriate)
- [Y] I have added documentation (where appropriate)
- [-] Any dependent changes have already been merged

### CODE REVIEW

File: frontend/src/modules/assetManagement/templates/assetManagementModalNewAsset.hbs [hunk 1]
```diff
@@ -38,6 +38,46 @@
...
+                <option value="CC-BY-NC">CC-BY-NC</option>
+                <option value="CC-BY-NC-SA">CC-BY-NC-SA</option>
+                <option value="CC-BY-ND">CC-BY-ND</option>
+                <option value="CC-BY-ND">CC-BY-NC-ND</option>
+                <option value="All rights reserved">All rights reserved</option>
+                <option value="Shutterstock standard licence">Shutterstock standard licence</option>
+            </select>
+            <script>
+            $( document ).ready(function() {
+               $('.asset-licence').val('{{ licence }}');
```
Comment (lc-thomasberger) [inline frontend/src/modules/assetManagement/templates/assetManagementModalNewAsset.hbs]: Can you please try to put this code inside the view that renders the template.

File: frontend/src/modules/assetManagement/templates/assetManagementModalNewAsset.hbs [hunk 2]
```diff
@@ -38,6 +38,46 @@
...
+            <textarea class="asset-source required" name="source" placeholder="{{t 'app.assetsource'}}">{{ source }}</textarea>
+        </div>
+
+        <div class="form-group">
+            <label for="licence">
+                {{t 'app.assetlicence'}} <span class="req">*</span>
+                <span class="error"></span>
+            </label>
+            <select class="asset-licence required" name="licence" >
+                <option value=""></option>
```
Comment (lc-thomasberger) [inline frontend/src/modules/assetManagement/templates/assetManagementModalNewAsset.hbs]: If we use select we can't have "custom" license types.

File: frontend/src/modules/assetManagement/views/assetManagementModalNewAssetView.js [hunk 3]
```diff
@@ -83,14 +83,17 @@ define(function(require){
...
+      var licence = this.$('.asset-licence').val();
         // If model is new then uploadFile
         if (this.model.isNew()) {
           this.uploadFile();
           // Return false to prevent the page submitting
           return false;
         } else {
           // Else just update the title, description and tags
-          this.model.set({title: title, description: description});
+          this.model.set({title: title, description: description, attribution: attribution, source: source, licence: licence});
```
Comment (lc-thomasberger) [inline frontend/src/modules/assetManagement/views/assetManagementModalNewAssetView.js]: Can you please make this multi-line.

File: frontend/src/modules/assetManagement/templates/assetManagementModalNewAsset.hbs (hunk 2 above)
Comment (davetaz) [reply frontend/src/modules/assetManagement/templates/assetManagementModalNewAsset.hbs]: Would there be a better way to define a set of values someone can customise that can be loaded here? Something in properties.schema?

Comment (lc-thomasberger) [reply frontend/src/modules/assetManagement/templates/assetManagementModalNewAsset.hbs]: Not really as far as I know. Maybe leave the dropdown but also add a text-input to allow custom text or only use the custom text-input.

File: frontend/src/modules/assetManagement/templates/assetManagementModalNewAsset.hbs (hunk 1 above)
Comment (davetaz) [reply frontend/src/modules/assetManagement/templates/assetManagementModalNewAsset.hbs]: I'm not entirely sure how to pull the current value out in the template. I put this here as the template seems to fill in the values for everything else. Help welcome

File: frontend/src/modules/assetManagement/views/assetManagementModalNewAssetView.js (hunk 3 above)
Comment (davetaz) [reply frontend/src/modules/assetManagement/views/assetManagementModalNewAssetView.js]: done

**Pull request closed without merging**